### ISSUE TITLE
Add `ifThenElse` constructor for `HttpMiddleware`

### ISSUE THREAD

**Issue by tusharmath** [zio/zio-http#454]
**Sample Signature**
```scala
val app1: HttpApp[Any, Nothing] = ???
val app2: HttpApp[Any, Nothing] = ???
val app3: HttpApp[Any, Nothing] = ???

val condM: (Method, Url, List[Header]) => UIO[Boolean]
val cond: (Method, Url, List[Header]) => Boolean

val app = app1 @@ ifThenElseM(condM)(app2, app3)
val app = app1 @@ ifThenElse(cond)(app2, app3)
```
Conditionally switches the control from one app to another.

**Comment by pravin-raha** [comment]
I will work on it

**Comment by tusharmath** [comment]
fixed in #485
